File: backend/src/BiteRight.Application/Commands/Products/Edit/EditRequest.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: EditRequest.cs
// # Author: Łukasz Sobczak
// # Created: 26-02-2024
// # ==============================================================================

#region

using System;
using BiteRight.Application.Dtos.Products;
using MediatR;
using Newtonsoft.Json;

#endregion

namespace BiteRight.Application.Commands.Products.Edit;

public class EditRequest : IRequest<EditResponse>
{
    public EditRequest(
        Guid productId,
        string name,
        string description,
        double? priceValue,
        Guid? priceCurrencyId,
        ExpirationDateKindDto expirationDateKind,
        DateOnly? expirationDate,
        Guid categoryId,
        double amountCurrentValue,
        double amountMaxValue,
        Guid amountUnitId
    )
    {
        ProductId = productId;
        Name = name;
        Description = description;
        PriceValue = priceValue;
        PriceCurrencyId = priceCurrencyId;
        ExpirationDateKind = expirationDateKind;
        ExpirationDate = expirationDate;
        CategoryId = categoryId;
        AmountCurrentValue = amountCurrentValue;
        AmountMaxValue = amountMaxValue;
        AmountUnitId = amountUnitId;
    }

    [JsonIgnore]
    public Guid ProductId { get; set; }

    public string Name { get; set; }
    public string Description { get; set; }
    public double? PriceValue { get; set; }
    public Guid? PriceCurrencyId { get; set; }
    public ExpirationDateKindDto ExpirationDateKind { get; set; }
    public DateOnly? ExpirationDate { get; set; }
    public Guid CategoryId { get; set; }
    public double AmountCurrentValue { get; set; }
    public double AmountMaxValue { get; set; }
    public Guid AmountUnitId { get; set; }
}

File: mobile/app/src/main/java/com/sobczal2/biteright/data/api/requests/products/EditRequest.kt
package com.sobczal2.biteright.data.api.requests.products

import com.google.gson.annotations.SerializedName
import com.sobczal2.biteright.data.api.common.Exclude
import com.sobczal2.biteright.dto.products.ExpirationDateKindDto
import java.time.LocalDate
import java.util.UUID

data class EditRequest(
    @Exclude val productId: UUID,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String,
    @SerializedName("priceValue") val priceValue: Double?,
    @SerializedName("priceCurrencyId") val priceCurrencyId: UUID?,
    @SerializedName("expirationDateKind") val expirationDateKind: ExpirationDateKindDto,
    @SerializedName("expirationDate") val expirationDate: LocalDate?,
    @SerializedName("categoryId") val categoryId: UUID,
    @SerializedName("amountCurrentValue") val amountCurrentValue: Double,
    @SerializedName("amountMaxValue") val amountMaxValue: Double,
    @SerializedName("amountUnitId") val amountUnitId: UUID
)